src/Struct/FindologicProduct.php

```diff
@@ -136,7 +136,8 @@ public function __construct(
         ContainerInterface $container,
         string $shopkey,
         array $customerGroups,
-        Item $item
+        Item $item,
+        ?Config $config = null
```

Comment:
This comment is still open, without any code change.

Reply:
Fetching services from container is not ideal. It is always recommended to use DI instead of container. We would have to mock the whole container just to pass our Config mock. Besides, even if we set the mocked config in container, it does not work, because the sales channel already initializes the config service so it cannot be replaced. 
If we reset it in tearDown it gives the following error in the next test case which needs to use the container. `The previous test case's transaction was not closed properly.` 

the functionality does not really gets affected with this additional argument because we are setting it default as null which in turn always gets from the already initialized container in a real scenario. But in the test cases it makes things much simpler rather than mocking the whole container and all the services it calls. I just do not see a good reason to use the container for getting services, and shopware itself does not recommend that.

Reply:
Fine, I took this point for our future major 3.x release.